src/oak_speech.c

```diff
@@ -821,403 +1035,407 @@ static void Task_OakSpeech7(u8 taskId)
         CopyBgTilemapBufferToVram(1);
-        DestroyLinkedPikaOrGrassPlatformSprites(taskId, 0);
-        data[3] = 80;
-        gTasks[taskId].func = Task_OakSpeech9;
+        DestroyPikachuOrPlatformSprites(taskId, SPRITE_TYPE_PIKACHU);
+        tTimer = 80;
+        gTasks[taskId].func = Task_OakSpeech_Init;
     }
 }
 
-static void Task_OakSpeech9(u8 taskId)
+static void Task_OakSpeech_Init(u8 taskId)
 {
-    s16 * data = gTasks[taskId].data;
+    s16 *data = gTasks[taskId].data;
     u32 size = 0;
-
-    if (data[3] != 0)
-        data[3]--;
+    if (tTimer != 0)
+    {
+        tTimer--;
+    }
     else
     {
-        sOakSpeechResources->solidColorsGfx = MallocAndDecompress(sOakSpeechGfx_SolidColors, &size);
-        LoadBgTiles(1, sOakSpeechResources->solidColorsGfx, size, 0);
-        CopyToBgTilemapBuffer(1, sOakSpeech_BackgroundTilemap, 0, 0);
+        sOakSpeechResources->oakSpeechBackgroundTiles = MallocAndDecompress(sOakSpeech_Background_Tiles, &size);
+        LoadBgTiles(1, sOakSpeechResources->oakSpeechBackgroundTiles, size, 0);
+        CopyToBgTilemapBuffer(1, sOakSpeech_Background_Tilemap, 0, 0);
         CopyBgTilemapBufferToVram(1);
         CreateNidoranFSprite(taskId);
-        LoadOaksSpeechTrainerPic(OAK_PIC, 0);
-        CreatePikaOrGrassPlatformSpriteAndLinkToCurrentTask(taskId, 1);
+        LoadTrainerPic(OAK_PIC, 0);
+        CreatePikachuOrPlatformSprites(taskId, SPRITE_TYPE_PLATFORM);
         PlayBGM(MUS_ROUTE24);
         BeginNormalPaletteFade(PALETTES_ALL, 5, 16, 0, RGB_BLACK);
-        data[3] = 80;
+        tTimer = 80;
         ShowBg(2);
-        gTasks[taskId].func = Task_OakSpeech10;
+        gTasks[taskId].func = Task_OakSpeech_WelcomeToTheWorld;
     }
 }
 
-#define OaksSpeechPrintMessage(str, speed) ({ \
-    DrawDialogueFrame(0, FALSE);\
-    if (str != gStringVar4) \
-    { \
-        StringExpandPlaceholders(gStringVar4, str); \
+#define OakSpeechPrintMessage(str, speed) ({                                                                                                 \
+    DrawDialogueFrame(0, FALSE);                                                                                                             \
+    if (str != gStringVar4)                                                                                                                  \
+    {                                                                                                                                        \
+        StringExpandPlaceholders(gStringVar4, str);                                                                                          \
         AddTextPrinterParameterized2(0, FONT_MALE, gStringVar4, speed, NULL, TEXT_COLOR_DARK_GRAY, TEXT_COLOR_WHITE, TEXT_COLOR_LIGHT_GRAY); \
-    } \
-    else \
-    { \
-        AddTextPrinterParameterized2(0, FONT_MALE, str, speed, NULL, TEXT_COLOR_DARK_GRAY, TEXT_COLOR_WHITE, TEXT_COLOR_LIGHT_GRAY); \
-    } \
-    CopyWindowToVram(0, COPYWIN_FULL); \
```

Comment:
The `0`s in this macro are the window id you defined `WIN_INTRO_TEXTBOX` for (same for the associated calls to `IsTextPrinterActive` and `ClearDialogWindowAndFrame`)